components/sidebar/sidebar_channel/sidebar_channel_close/index.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
+
+import {getCurrentChannelId, shouldHideDefaultChannel} from 'mattermost-redux/selectors/entities/channels';
+import {Channel} from 'mattermost-redux/types/channels';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+import {isFavoriteChannel} from 'mattermost-redux/utils/channel_utils';
+
+import Constants from 'utils/constants';
+
+import SidebarChannelClose from './sidebar_channel_close';
+
+type OwnProps = {
+    channel: Channel;
+    show: boolean;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const currentChannelId = getCurrentChannelId(state);
+    const active = ownProps.channel.id === currentChannelId;
+
+    let shouldHideChannelClose = false;
+    if (
+        ownProps.channel.name === Constants.DEFAULT_CHANNEL &&
+        !active &&
+        shouldHideDefaultChannel(state, ownProps.channel) &&
+        !isFavoriteChannel(state.entities.preferences.myPreferences, ownProps.channel.id)
+    ) {
+        shouldHideChannelClose = true;
+    }
```

Comment:
What's this logic for exactly? Perhaps we could refactor it into a helper function or something so that it's easier to tell what's going on here.

Reply:
From the connector for `sidebar_channel.jsx`:
```
let shouldHideChannel = false;
if (
    channel.name === Constants.DEFAULT_CHANNEL &&
    !ownProps.active &&
    shouldHideDefaultChannel(state, channel) &&
    !isFavoriteChannel(state.entities.preferences.myPreferences, channel.id)
) {
    shouldHideChannel = true;
}
```

This long is meant to hide the default channel (ie. Town Square) when `ExperimentalHideTownSquareinLHS` is on. Is this logic something we're keeping?
Also, I'm not sure how to refactor it other than pushing it down into redux.

Reply:
If it's for that setting, why is it in this component? `shouldHideChannelClose` sounds like it's hiding the close button on the channel, not the channel itself.

Also, we still need to discuss whether or not we're keeping those settings, but that's something we need UX and PM buy-in for

Reply:
I can change the name if that would make it clearer, but at some point some component needs to have this logic. Since it's only really related to whether you can close the channel or not, I don't see any point in having it further up the chain.